### DISCUSSION TITLE
labgrid-exporter dies and client throws the error when serial connection is disconnected

### DISCUSSION THREAD

**Discussion by ekathva** [labgrid-project/labgrid#821] [Q&A]
My hardware setup:
[Raspi <----tty console ---->(/dev/ttyUSB0 at server running exporter and coordinator) <---dev laptop--->]

System setup:
I have ser2net service running - which maps the local /dev/ttyUSB0 to tcp port 20410 - my exporter configuration has NetworkSerialPort exported at port 20410, host: <server>, speed: 115200. This resource is added to place rpi_4. From laptop, I can use labgrid-client console command which gets me into the Raspi (so, it is working as expected). Now, in my use case, I am planning to upgrade the device with sysupgrade (which reboots the Raspi). So, in this case, if I reboot the Raspi, I get the error:

Error:
- client:
```
ERROR: Task exception was never retrieved
future: <Task finished name='Task-4' coro=<ClientSession.onJoin() done, defined at /home/ekathva/.env/wrt-test/lib/python3.9/site-packages/labgrid-0.3.1.dev151+ge9ffb93.d20210819-py3.9.egg/labgrid/remote/client.py:76> exception=ApplicationError('no callee registered for procedure <org.labgrid.coordinator.get_resources>')>
Traceback (most recent call last):
  File "/home/ekathva/.env/wrt-test/lib/python3.9/site-packages/labgrid-0.3.1.dev151+ge9ffb93.d20210819-py3.9.egg/labgrid/remote/client.py", line 78, in onJoin
    resources = await self.call(
autobahn.wamp.exception.ApplicationError: ApplicationError(error=<wamp.error.no_such_procedure>, args=['no callee registered for procedure <org.labgrid.coordinator.get_resources>'], kwargs={}, enc_algo=None, callee=None, callee_authid=None, callee_authrole=None, forward_for=None)
```
Coordinator:
I have the journalctl -u labgrid-coordinator.service -f (which outputs nothing)

Exporter:
- systemctl restart <exporter.service> fails (unless I restart the coordinator also)
- In the following log, wait.sh is script wrapper, which waits until labgrid-coordinator is up and running and then starts the exporter service.
- errors:
```
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:12 Auto ping/pong: sending ping auto-ping/pong
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:12 Expecting ping in 5.0 seconds for auto-ping/pong
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:17 dropping connection to peer tcp4:127.0.0.1:20408 with abort=True: WebSocket ping timeout (peer did not respond with pong in time)
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:17 _connectionLost: None
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:17 WAMP-over-WebSocket transport lost: wasClean=False, code=1006, reason="connection was closed uncleanly (WebSocket ping timeout (peer did not respond with pong in time))"
Aug 26 14:31:17 server1 wait.sh[13939]: connection lost
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:17 session closed with reason wamp.close.transport_lost [WAMP transport was lost without closing the session 4707650053822370 before]
Aug 26 14:31:17 server1 wait.sh[13939]: 2021-08-26T14:31:17 Close <_UnixSelectorEventLoop running=False closed=False debug=True>
Aug 26 14:31:17 server1 systemd[1]: labgrid-exporter.service: Main process exited, code=exited, status=100/n/a
Aug 26 14:31:17 server1 systemd[1]: labgrid-exporter.service: Failed with result 'exit-code'.
```

Is there any way I can fix it? Or I just always restart coordinator, exporter every time my Raspi gets rebooted after sysupgrade?

Thank you :)

**Comment by Emantor** [comment]
To clarify this, the exporter is not running on the rpi4, right? I would investigate why power switching the rpi also seems to kill the network connection to your server.
